### PR TITLE
Introducing oneDAL roles for contributors and maintainers

### PR DESCRIPTION
**Pull request by napetrov**
# Description
This is initial version of MAINTAINERS.md that define roles and list certain individuals in those roles. 

Feedback and recommendations are welcome

### CODE REVIEW

**Comment by napetrov** [comment]
@keeranroth @rakshithgb-fujitsu  - please take a look on this. open for recommendations/suggestions

**Comment by keeranroth** [comment]
@napetrov thanks for the mention, and I'm happy to be down as a code owner for the RISC-V part of the code base, with the caveats listed below.

The wording is good, except I'm always hesitant about putting time commitments on roles, although it would be nice to show that some effort is expected here. In general, as long as a person is fulfilling their responsibilities, then the time commitment doesn't play too much of a role. Using me as an example, for the RISC-V part, I am happy to be down as a code owner, and will have a quick turnaround on code review. However, that won't take 25% of my time, for what is a relatively new and not yet widely used part of the oneDAL project. I can make fixes to broken pipelines, but turnaround might be on the order of a week or two, depending on other responsibilities. As RISC-V usage and contributions pick up, it might be good to come back to this and choose a more active member of the community as a code owner.

So, I suggest that the bullet point around the time requirement be taken out. For the code owners in RISC-V section, I'm happy to be listed, as long as the caveat mentioned above is acceptable. What do you think?

**Comment by rakshithgb-fujitsu** [comment]
I agree with Keeran on this; as the oneDAL community expands, more individuals will naturally take on roles as code owners and maintainers. Implementing a time requirement isn’t a beneficial approach.

**Comment by napetrov** [comment]
@keeranroth  @rakshithgb-fujitsu  Thanks for feedback!   cleaned up those lines on time commitments,

**Review by keeranroth — APPROVED** [review]
Looks good to me. Thanks @napetrov

**Review by icfaust — APPROVED** [review]
@napetrov ready to merge, though we just double check on the CODEOWNERS change, GitHub is complaining about the non-public nature of ```@oneapi-src/owner-onedal```. Follow up task? 

Will mention uxlfoundation/open-source-working-group#14 just to properly link to the issue.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -26,3 +26,10 @@ cpp/           @Alexsandruss @samir-nasibli @Alexandr-Solovev
 
 # Tree based methods
 dtrees         @razdoburdin @ahuber21 @avolkov-intel @icfaust
+
+# Governance and process
+/.github/CODEOWNERS @oneapi-src/owner-onedal
+/SECURITY.md @oneapi-src/owner-onedal
+/MAINTAINERS.md @oneapi-src/owner-onedal
+/CONTRIBUTING.md @oneapi-src/owner-onedal
+/CODE_OF_CONDUCT.md @oneapi-src/owner-onedal
```
Comment (napetrov) [inline .github/CODEOWNERS:35]:
```suggestion
/.github/CODEOWNERS @oneapi-src/oneDAL-maintainer
/SECURITY.md @oneapi-src/oneDAL-maintainer
/MAINTAINERS.md @oneapi-src/oneDAL-maintainer
/CONTRIBUTING.md @oneapi-src/oneDAL-maintainer
/CODE_OF_CONDUCT.md @oneapi-src/oneDAL-maintainer
```

**Pull request merged**